cognite/client/data_classes/transformations.py

```diff
@@ -0,0 +1,261 @@
+from cognite.client.data_classes._base import *
+from cognite.client.data_classes.transformation_jobs import TransformationJob, TransformationJobList
+from cognite.client.data_classes.transformation_schedules import TransformationSchedule
+from cognite.client.data_classes.transformation_schema import TransformationSchemaColumnList
+from cognite.client.data_classes.transformations_common import *
+
+
+class Transformation(CogniteResource):
+    """The transformations resource allows transforming data in CDF.
+
+    Args:
+        id (int): A server-generated ID for the object.
+        external_id (str): The external ID provided by the client. Must be unique for the resource type.
+        name (str): The name of the Transformation.
+        query (str): SQL query of the transformation.
+        destination (TransformationDestination): see TransformationDestination for options.
+        conflict_mode (str): What to do in case of id collisions: either "abort", "upsert", "update" or "delete"
+        is_public (bool): Indicates if the transformation is visible to all in project or only to the owner.
+        ignore_null_fields (bool): Indicates how null values are handled on updates: ignore or set null.
+        source_api_key (str): Configures the transformation to authenticate with the given api key on the source.
+        destination_api_key (str): Configures the transformation to authenticate with the given api key on the destination.
+        source_oidc_credentials (Optional[OidcCredentials]): Configures the transformation to authenticate with the given oidc credentials key on the destination.
+        destination_oidc_credentials (Optional[OidcCredentials]): Configures the transformation to authenticate with the given oidc credentials on the destination.
+        created_time (int): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
+        last_updated_time (int): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
+        owner (str): Owner of the transformation: requester's identity.
+        owner_is_current_user (bool): Indicates if the transformation belongs to the current user.
+        has_source_api_key (bool): Indicates if the transformation is configured with a source api key.
+        has_destination_api_key (bool): Indicates if the transformation is configured with a destination api key.
+        has_source_oidc_credentials (bool): Indicates if the transformation is configured with a source oidc credentials set.
+        has_destination_oidc_credentials (bool): Indicates if the transformation is configured with a destination oidc credentials set.
```

Comment:
do we need these all?

Reply:
I think this is huge because we use the same type for TransformationCreate and TransformationRead

Reply:
`owner`, `owner_is_current_user`, `isPublic`, `source_api_key`, `destination_api_key`, `has_source_api_key` and `has_destination_api_key` are valid when working with API keys.  Hope that we can remove them after going v1 when API keys become deprecated fully.